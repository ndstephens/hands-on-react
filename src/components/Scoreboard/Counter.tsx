import styled from '@emotion/styled';

interface Props {
  /**
   * Number in the counter
   */
  children: string;
}

export const Counter = ({ children }: Props) => {
  return <Frame>{children}</Frame>;
};

const Frame = styled.div`
  display: inline-block;
  padding: 0 0.3vw;
  color: #ec433c;
  border: 0.15vw inset;
  /* line-height: 2vw; */
  line-height: 1.6;
  letter-spacing: 0.08em;
  background-color: #333;
  text-shadow: 0 0 0.1vw currentColor;
`;
