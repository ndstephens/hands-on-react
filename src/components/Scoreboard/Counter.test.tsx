import { render } from '@testing-library/react';

import { Counter } from './Counter';

it('Counter renders correctly', () => {
  const { asFragment } = render(<Counter>001</Counter>);

  expect(asFragment()).toMatchSnapshot();
});
