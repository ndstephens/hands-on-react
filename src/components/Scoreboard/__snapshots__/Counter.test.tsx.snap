// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1tfs0js"
  >
    001
  </div>
</DocumentFragment>
`;
